tools/runners/run-mongo.js

```diff
@@ -213,8 +213,7 @@ if (process.platform === 'win32') {
             var foundPort = parseInt(m[2], 10);
             var foundPath = m[3];
 
-            if ( (! port || port === foundPort) &&
-                 (! dbDir || dbDir === foundPath)) {
+            if (! dbDir || dbDir === foundPath) {
```

Comment:
@tmeasday The `findMongoPids(null, port)` version is broken because we pass the new `port`, though `mongod` is still running on the old port, which we don't know. There seems to be an old assumption that `mongod` can run with the same `dbPath` but different ports, so if you kill the one running on the port you're about to use, then all's well. That's just not true anymore, if it ever was.